block-storage/src/main/scala/coop/rchain/blockstorage/ApprovedStore.scala

```diff
@@ -0,0 +1,20 @@
+package coop.rchain.blockstorage
+
+import cats.effect.Sync
+import cats.syntax.all._
+import coop.rchain.blockstorage.dag.codecs.{codecApprovedBlock, codecBlockHash}
+import coop.rchain.casper.protocol.ApprovedBlock
+import coop.rchain.models.BlockHash.BlockHash
+import coop.rchain.shared.syntax._
+import coop.rchain.store.{KeyValueStoreManager, KeyValueTypedStore}
+
+object ApprovedStore {
+  type ApprovedStore[F[_]] = KeyValueTypedStore[F, BlockHash, ApprovedBlock]
+
+  def apply[F[_]: Sync](
+      kvm: KeyValueStoreManager[F]
+  ): F[KeyValueTypedStore[F, BlockHash, ApprovedBlock]] =
+    kvm
+      .store("blocks-approved")
+      .map(_.toTypedStore[BlockHash, ApprovedBlock](codecBlockHash, codecApprovedBlock))
```

Comment:
Use `bytes` scodec as a key. No need to involve ByteString when we know the key is `Array[Byte]` which can easily be converted to scodec.

Reply:
@tgrospic, I [simplified](https://github.com/rchain/rchain/pull/3655/commits/a5176fc33d155fce6725a71e09987e076ffd5339) ApprovedStore key to just 1 byte. Is it ok?